include/seqan3/range/views/take_until.hpp

```diff
@@ -465,6 +204,312 @@ class view_take_until : public std::ranges::view_interface<view_take_until<urng_
+    {
+        return rhs == lhs;
+    }
+
+    //!\brief Return the saved at_end state.
+    bool operator!=(sentinel_type const & rhs) const
+        noexcept(noexcept(std::declval<basic_consume_iterator &>() == rhs))
+    {
+        return !(*this == rhs);
+    }
+
+    //!\brief Return the saved at_end state.
+    friend bool operator!=(sentinel_type const & lhs, basic_consume_iterator const & rhs)
+        noexcept(noexcept(rhs != lhs))
+    {
+        return rhs != lhs;
+    }
+    //!\}
+};
+
+//!\brief The sentinel type of take_until, provides the comparison operators.
+template <std::ranges::view urng_t, typename fun_t, bool or_throw, bool and_consume>
+template <bool is_const_range>
+class view_take_until<urng_t, fun_t, or_throw, and_consume>::basic_sentinel
+{
+private:
+    //!\brief The base type of the underlying range.
+    using urng_base_type = std::conditional_t<is_const_range, std::add_const_t<urng_t>, urng_t>;
+    //!\brief The sentinel type of the underlying range.
+    using urng_sentinel_type = std::ranges::sentinel_t<urng_base_type>;
+    //!\brief Auxiliary type.
+    using predicate_ref_t = std::conditional_t<is_const_range,
+                                               std::remove_reference_t<fun_t> const &,
+                                               std::remove_reference_t<fun_t> &>;
+
+    //!\brief The actual end of the underlying range.
+    urng_sentinel_type urng_sentinel{};
+
+    //!\brief Reference to the predicate stored in the view.
+    ranges::semiregular_box_t<predicate_ref_t> predicate{};
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    basic_sentinel() = default; //!< Defaulted.
+    basic_sentinel(basic_sentinel const &) = default; //!< Defaulted.
+    basic_sentinel(basic_sentinel &&) = default; //!< Defaulted.
+    basic_sentinel & operator=(basic_sentinel const &) = default; //!< Defaulted.
+    basic_sentinel & operator=(basic_sentinel &&) = default; //!< Defaulted.
+    ~basic_sentinel() = default; //!< Defaulted.
+
+    /*!\brief Construct from a sentinel and a predicate.
+     * \param[in] urng_sentinel  The actual end of the underlying range.
+     * \param[in] predicate      Reference to the predicate stored in the view.
+     */
+    explicit basic_sentinel(urng_sentinel_type urng_sentinel, predicate_ref_t predicate) :
+        urng_sentinel{std::move(urng_sentinel)},
+        predicate{predicate}
+    {}
+
+    //!\brief Construct from a not const range a const range.
+    basic_sentinel(basic_sentinel<!is_const_range> other)
+        requires is_const_range && std::convertible_to<std::ranges::sentinel_t<urng_t>, urng_sentinel_type>
+        : urng_sentinel{std::move(other.urng_sentinel)},
+          predicate{other.predicate}
+    {}
+    //!\}
+
+    /*!\name Comparison operators
+     * \brief We define comparison against the sentinel.
+     * \{
+     */
+
+    //!\brief Evaluate functor, possibly throw.
```

Comment:
```suggestion
    /*!\brief Compares `lhs` with `rhs` for equality.
     * 
     * \details
     *
     * Invokes the stored predicate with the pointed-to element from `lhs` and returns the result of this evaluation.
     * If the end of the underlying range was reached before the predicate evaluated to `true` and the view has 
     * `or_throw` enabled, then a seqan3::unexpected_end_of_input exception will be thrown, otherwise the returned
     * value is `true`.
     */
```